doc/rbd/rbd-mirroring.rst

```diff
@@ -0,0 +1,233 @@
+(available from the ``rbd mirror pool info`` command)::
+
+        rbd mirror pool peer remove {pool-name} {peer-uuid}
+
+For example::
+
+        rbd --cluster local mirror pool peer remove image-pool 55672766-c02b-4729-8567-f13a66893445
+        rbd --cluster remote mirror pool peer remove image-pool 60c0e299-b38f-4234-91f6-eed0a367be08
+
+Image Configuration
+===================
+
+Unlike pool configuration, image configuration only needs to be performed against
+a single Ceph cluster (i.e. the primary data center Ceph cluster).
+
+Enable Image Journaling Support
+-------------------------------
+
+RBD mirroring uses the RBD journaling feature to ensure that the replicated
+image always remains crash-consistent.  Before an image can be mirrored to
+a peer cluster, the journaling feature must be enabled.  The feature can be
+enabled at image creation time by including ``--image-feature exclusive-lock,journaling``
+option to the ``rbd`` command. 
+
+Alternatively, the journaling feature can be dynamically enabled on
+pre-existing RBD images.  To enable journaling with ``rbd``, specify
+the ``feature enable`` command, the pool and image name, and the feature name::
+
+        rbd feature enable {pool-name}/{image-name} {feature-name}
+
+For example::
+
+        rbd --cluster local feature enable image-pool/image-1 journaling
+
+.. note:: The journaling feature is dependent on the exclusive-lock feature.  If
+   the exclusive-lock feature is not already enabled, it should be enabled prior
+   to enabling the journaling feature.
+
+.. tip:: You can enable journaling on all new images by default by adding
+   ``rbd default features = 125`` to your Ceph configuration file.
+
+Enable Image Mirroring
+----------------------
+
+If the mirroring is configured in ``image`` mode for the image's pool, then it
+is necessary to explicitly enable mirroring for each image within the pool.
+To enable mirroring for a specific image with ``rbd``, specify the
+``mirror image enable`` command along with the pool and image name::
+
+        rbd mirror image enable {pool-name}/{image-name}
+
+For example::
+
+        rbd --cluster local mirror image enable image-pool/image-1
+
+Disable Image Mirroring
+-----------------------
+
+To disable mirroring for a specific image with ``rbd``, specify the
+``mirror image disable`` command along with the pool and image name::
+
+        rbd mirror image disable {pool-name}/{image-name}
+
+For example::
+
+        rbd --cluster local mirror image disable image-pool/image-1
+
+Image Promotion and Demotion
+----------------------------
+
+An image can only be primary within a single Ceph cluster.  Attempting to
+access a non-primary image will result in an error.  In a failover scenario,
+access to the primary image should be stopped (e.g. power down the VM or remove
+the associated drive from a VM), demote the current primary image, promote
+the new primary image, and resume access to the image on the alternate
+cluster.
```

Comment:
RBD cannot coordinate failover -- way too many moving parts outside our control.  For example, Cinder failover support in OpenStack will coordinate all of this (eventually) by disconnecting the drive, demoting, promoting, and reconnecting the drive.